tests/PHPUnit/Unit/DataAccess/LogQueryBuilder/JoinGeneratorTest.php

```diff
@@ -102,6 +102,34 @@ public function test_generate_getJoinString_manuallyJoinedAlready()
         $this->assertEquals($expected, $generator->getJoinString());
     }
 
+    public function test_generate_getJoinString_manuallyJoinedAlreadyWithCustomConditionInArray()
+    {
+        $generator = $this->generate(array(
+            'log_visit',
+            array('table' => 'log_conversion', 'joinOn' => 'log_visit.idvisit2 = log_conversion.idvisit2'),
+            'log_conversion'
+        ));
```

Comment:
I know, I tried to implement it but won't be trivial. We could create a new issue for it. Eg when array is like 

```php
array(
            'log_visit',
            'log_conversion',
            'log_link_visit_action',
            array('table' => 'log_conversion', 'joinOn' => 'log_link_visit_action.idvisit2 = log_conversion.idvisit2'),
```

Then I can't simply use the join condition of the second `log_conversion` array on the first `log_conversion` table since it might reference a table that is only joined later. Also I cannot remove the first `log_conversion` since it might be used in a join before the other `log_conversion` comes.